cyfra-core/src/main/scala/io/computenode/cyfra/core/layout/LayoutStruct.scala

```diff
@@ -0,0 +1,67 @@
+package io.computenode.cyfra.core.layout
+
+import io.computenode.cyfra.core.buffer.BufferRef
+import io.computenode.cyfra.dsl.Value
+import io.computenode.cyfra.dsl.buffer.GBuffer
+import izumi.reflect.Tag
+import izumi.reflect.macrortti.LightTypeTag
+
+import scala.compiletime.{error, summonAll}
+import scala.deriving.Mirror
+import scala.quoted.{Expr, Quotes, Type}
+
+case class LayoutStruct[T <: Layout: Tag](private[cyfra] val layoutRef: T, private[cyfra] val elementTypes: List[Tag[? <: Value]])
+
+object LayoutStruct:
+
+  inline given derived[T <: Layout: Tag]: LayoutStruct[T] = ${ derivedImpl }
+
+  def derivedImpl[T <: Layout: Type](using quotes: Quotes): Expr[LayoutStruct[T]] =
```

Comment:
I am not convinced this has to be macro. What I see you doing here is:
* extracting types of members 
* summoning stuff for those types

This can be done with case class and `Mirror`